x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -744,13 +745,17 @@ private void innerRefresh(String tokenDocId, Authentication userAuth, ActionList
                             try (StreamInput in = StreamInput.wrap(Base64.getDecoder().decode(authString))) {
                                 in.setVersion(authVersion);
                                 Authentication authentication = new Authentication(in);
-                                UpdateRequest updateRequest =
+                                UpdateRequestBuilder updateRequest =
                                     client.prepareUpdate(SecurityIndexManager.SECURITY_INDEX_NAME, TYPE, tokenDocId)
-                                        .setVersion(response.getVersion())
                                         .setDoc("refresh_token", Collections.singletonMap("refreshed", true))
-                                        .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL)
-                                        .request();
-                                executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN, updateRequest,
+                                        .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL);
+                                if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_6_7_0)) {
+                                    updateRequest.setIfSeqNo(response.getSeqNo());
+                                    updateRequest.setIfPrimaryTerm(response.getPrimaryTerm());
+                                } else {
+                                    updateRequest.setVersion(response.getVersion());
+                                }
```

Comment:
Small suggestion to make the condition tight proof (an assert could also work).
```
if (clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.V_6_7_0)
  && response.getSeqNo() != UNASSIGNED_SEQ_NO) {
    updateRequest.setIfSeqNo(response.getSeqNo());
    updateRequest.setIfPrimaryTerm(response.getPrimaryTerm());
} else {
    updateRequest.setVersion(response.getVersion());
}
```
There might be guarantees in other places in the code, I haven't looked for it thoroughly, you know better.

Reply:
@albertzaharovits sorry,  I didn't see this version of the comment before merging. I'll add it in a follow up.